cognite/powerops/resync/utils.py

```diff
@@ -113,9 +114,9 @@ def ext_id_factory(domain_model_type: type, data: dict) -> str:
 
     # TODO proper naming
     if domain_model_type is v1_data_classes.GeneratorEfficiencyCurveWrite:
-        return f"{type_prefix}_foo_{random.random()}"
+        return f"{type_prefix}:{_get_short_hash(data)}"
     elif domain_model_type is v1_data_classes.TurbineEfficiencyCurveWrite:
-        return f"{type_prefix}_foo_{random.random()}"
+        return f"{type_prefix}:{_get_short_hash(data)}"
```

Comment:
👍  Still not beautiful, but a real improvement

Reply:
Yea not sure how the `foo` snuck past everyone but at least now with using the short hash of the data I get a consistent external id every time it's run unless the content of the curve changes which is fine to get a new id I think since it immediately gets relinked in all relevant places to the new name.

Reply:
Best would be to include the generator name, and also the ref. value for turbine efficiency curves, but getting the generator name into this context is a bigger change, if I remember correctly, and I don't think we have to do that (now).

We don't have a mechanism to remove any old curves if content (and thereby hash) changes, right? (Don't have to handle that, but good to know.)